repos/system_upgrade/el8toel9/actors/opensslconfigcheck/libraries/opensslconfigcheck.py

```diff
@@ -0,0 +1,350 @@
+                'directive MinProtocol or MaxProtocol, which was previously applied to both '
+                'TLS and DTLS versions. This is no longer case and if you want to limit both '
+                'protocols, you need to add another option, for example for '
+                '`MinProtocol TLSv1.2` add the following line `DTLS.MinProtocol = DTLSv1.2`.'
+            ),
+            reporting.Severity(reporting.Severity.MEDIUM),
+            reporting.Tags([
+                    reporting.Tags.SECURITY,
+                    reporting.Tags.NETWORK,
+                    reporting.Tags.SERVICES
+            ]),
+        ] + resources)
+
+
+def check_duplicate_extensions(config):
+    dup = _openssl_duplicate_keys_in(config, "x509_extensions")
+    if dup:
+        reporting.create_report([
+            reporting.Title('There are duplicate x509 extensions defined in openssl.cnf'),
+            reporting.Summary(
+                'The OpenSSL configuration file `/etc/pki/tls/openssl.cnf` contains the '
+                'following duplicate X509 extensions: {}. With OpenSSL 3.0 only the last defined '
+                'will get used. Please, review the configuration file and remove duplicate '
+                'extensions to silence this warning.'.format(', '.join(dup))
+            ),
+            reporting.Severity(reporting.Severity.LOW),
+            reporting.Tags([
+                    reporting.Tags.SECURITY,
+                    reporting.Tags.NETWORK,
+                    reporting.Tags.SERVICES
+            ]),
+        ] + resources)
+
+
+def check_default_modules(config):
+    if config.openssl_conf != "default_modules" or not _openssl_find_block(config, "default_modules"):
+        reporting.create_report([
+            reporting.Title('Non-standard configuration of openssl.cnf'),
+            reporting.Summary(
+                'The OpenSSL configuration file `/etc/pki/tls/openssl.cnf` does not contain '
+                'expected initialization so it can not be updated to support OpenSSL 3.0 '
+                'with the new providers.'
+            ),
+            reporting.Remediation(
+                'The openssl.cnf file needs to contain the following initialization: '
+                '`openssl_conf = default_modules` and corresponding `[ default_modules] '
+                'block. The `openssl_conf` now contains {} or the `[ default_modules ]` '
+                'block is missing. '.format(config.openssl_conf)
+            ),
+            reporting.Flags([reporting.Flags.INHIBITOR]),
+            reporting.Severity(reporting.Severity.HIGH),
+            reporting.Tags([
+                    reporting.Tags.SECURITY,
+                    reporting.Tags.NETWORK,
+                    reporting.Tags.SERVICES
+            ]),
+        ] + resources)
+    else:
+        reporting.create_report([
+            reporting.Title('Updates to openssl.cnf'),
```

Comment:
@Jakuje just thinking about the iprovement of the title. wdyt?
```suggestion
            reporting.Title('The OpenSSL configuration will be updated to support OpenSSL 3.0'),
```

Reply:
This certainly sounds better. Fixed and squashed.